websites/A/Archive of Our Own/presence.ts

```diff
@@ -0,0 +1,60 @@
+const presence = new Presence({
+	clientId: "1005873313551220757",
+});
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: "https://archiveofourown.org/images/ao3_logos/logo_42.png",
+		},
+		[work, tag] = await Promise.all([
+			presence.getSetting<string>("work"),
+			presence.getSetting<string>("tag"),
+		]),
+		{ pathname, href } = document.location;
+	if (pathname === "/") presenceData.details = "Viewing home page";
+	else if (pathname.includes("/tags/")) {
+		if (tag) {
+			presenceData.details = `Browsing tag : ${
+				document.querySelector(".heading > .tag").textContent
+			}`;
+			presenceData.state = document.querySelector("h2.heading").textContent;
+		} else presenceData.details = " Browsing tags...";
+	} else if (pathname.includes("/works/")) {
+		if (work) {
+			presenceData.details = `Reading : ${
+				document.querySelector("h2").textContent
+			}`;
+
+			if (document.querySelector("div.chapter > h3"))
+				presenceData.state = "Oneshot";
+			else {
+				presenceData.state = `${
+					document.querySelector("div.chapter > h3").textContent
+				}`;
+			}
```

Comment:
What

Reply:
Really what
```suggestion
			presenceData.state = document.querySelector("div.chapter > h3") ? document.querySelector("div.chapter > h3").textContent : "Oneshot";
```

Reply:
And if we want to be extreme, you could store the query in a variable.